graph_weather/data/bufr_process.py

```diff
@@ -0,0 +1,515 @@
+
+        Returns:
+            -> xarray Dataset in NNJA-AI format
+        """
+        df = self.process_files_to_dataframe(filepath=filepath)
+
+        if df.empty:
+            logger.warning(f"No data to convert to xarray from {filepath}")
+            return xr.Dataset()
+
+        data_vars = {}
+        for col in df.columns:
+            if col not in ["OBS_TIMESTAMP", "LAT", "LON"]:
+                data_vars[col] = (["observation"], df[col].values)
+
+        ds = xr.Dataset(
+            data_vars=data_vars,
+            coords={
+                "obs": df.index,
+                "time": ("obs", df["OBS_TIMESTAMP"].values),
+                "lat": ("obs", df["LAT"].values),
+                "lon": ("obs", df["LON"].values),
+            },
+        )
+        ds.attrs["source"] = self.schema.source_name
+        ds.attrs["processing_timestamp"] = pd.Timestamp.now().isoformat()
+        ds.attrs["num_observations"] = len(df)
+
+        return ds
+
+    def process_files_to_parquet(self, filepath: str, output_path: str) -> None:
+        """
+        Process BUFR file and save as Parquet in NNJA-AI format.
+
+        Args:
+            -> filepath: Path to BUFR file
+            -> output_path: Path for output Parquet file
+        """
+        df = self.process_files_to_dataframe(filepath=filepath)
+
+        if not df.empty:
+            df.to_parquet(output_path, index=False)
+            logger.info(f"Saved {len(df)} observations to {output_path}")
+        else:
+            logger.warning(f"No data to save for {filepath}")
+
+
+class BUFR_dataloader:
+
+    SCHEMA_REGISTRY = {
+        "ADPUPA": ADPUPA_schema,
+        "CrIS": CRIS_schema,
+    }
+
+    def __init__(self, filepath: str, schema_name: Optional[str] = None):
+        """
+        Args:
+            filepath: Path to BUFR file or directory
+            schema_name: Data source name ('ADPUPA', 'CrIS', etc.)
+        """
+        self.filepath = Path(filepath)
+        self.schema_name = schema_name or self._infer_schema_from_path()
+
+        if self.schema_name not in self.SCHEMA_REGISTRY:
+            raise ValueError(
+                f'Unknown schema "{self.schema_name}". Available: {list(self.SCHEMA_REGISTRY.keys())}'
+            )
+
+        self.schema = self.SCHEMA_REGISTRY[self.schema_name]()
+        self.processor = BUFR_processor(self.schema)
+
+    def _infer_schema_from_path(self) -> str:
+        """Infer schema from filename or path patterns."""
+        filename = self.filepath.name.lower()
+
+        if "adpupa" in filename or "raob" in filename or "sound" in filename:
+            return "ADPUPA"
+        elif "cris" in filename:
+            return "CrIS"
+        elif "iasi" in filename:
+            return "IASI"
+        elif "atms" in filename:
+            return "ATMS"
+        else:
+            # Default to ADPUPA for now
+            logger.warning(f"Could not infer schema from {filename}, defaulting to ADPUPA")
```

Comment:
Don't default to one, if it doesn't match, then throw an error, silently changing to a different parser just makes things harder